--- a/gst/avi/gstavidemux.c
+++ b/gst/avi/gstavidemux.c
@@ -4747,12 +4747,16 @@
     buf = gst_avi_demux_invert (stream, buf);
 
     /* mark non-keyframes */
-    if (keyframe)
+    if (keyframe) {
       GST_BUFFER_FLAG_UNSET (buf, GST_BUFFER_FLAG_DELTA_UNIT);
-    else
+      GST_BUFFER_TIMESTAMP (buf) = timestamp;
+    }
+    else {
       GST_BUFFER_FLAG_SET (buf, GST_BUFFER_FLAG_DELTA_UNIT);
+      GST_BUFFER_TIMESTAMP (buf) = GST_CLOCK_TIME_NONE;
+    }
 
-    GST_BUFFER_TIMESTAMP (buf) = timestamp;
+    
     GST_BUFFER_DURATION (buf) = duration;
     GST_BUFFER_OFFSET (buf) = out_offset;
     GST_BUFFER_OFFSET_END (buf) = out_offset_end;
--- a/gst/isomp4/qtdemux.h
+++ b/gst/isomp4/qtdemux.h
@@ -111,7 +111,7 @@
   guint64 seek_offset;
 
   gboolean upstream_seekable;
-  gboolean upstream_size;
+  gint64 upstream_size;
 };
 
 struct _GstQTDemuxClass {
