--- a/configure.ac
+++ b/configure.ac
@@ -582,6 +582,7 @@
 ])
 dnl bit of a misnomer, but keep the conditional named like this so we don't
 dnl have to change too much elsewhere
+HAVE_CHECK=no
 AM_CONDITIONAL(HAVE_CHECK, test "x$BUILD_CHECK" = "xyes")
 
 dnl configure the desired buffer alignment
--- a/tools/gst-inspect.c
+++ b/tools/gst-inspect.c
@@ -1572,7 +1572,7 @@
   g_option_context_add_main_entries (ctx, options, GETTEXT_PACKAGE);
   g_option_context_add_group (ctx, gst_init_get_option_group ());
   if (!g_option_context_parse (ctx, &argc, &argv, &err)) {
-    g_print ("Error initializing: %s\n", err->message);
+    g_print ("Error initializing: %s\n", err ? err->message : "(null)");
     exit (1);
   }
   g_option_context_free (ctx);
--- a/gst/parse/grammar.y
+++ b/gst/parse/grammar.y
@@ -26,7 +26,6 @@
  */
 
 #define YYERROR_VERBOSE 1
-#define YYLEX_PARAM scanner
 
 #define YYENABLE_NLS 0
 
@@ -648,6 +647,7 @@
 %right '.'
 %left '!' '='
 
+%lex-param { void *scanner }
 %parse-param { void *scanner }
 %parse-param { graph_t *graph }
 %pure-parser
