--- a/charset/fribidi-char-sets-utf8.c
+++ b/charset/fribidi-char-sets-utf8.c
@@ -65,13 +65,22 @@ fribidi_utf8_to_unicode (
 	  *us++ = ((*s & 0x1f) << 6) + (*(s + 1) & 0x3f);
 	  s += 2;
 	}
-      else			/* 3 byte */
+      else if (ch <= 0xef)	/* 3 byte */
 	{
 	  *us++ =
 	    ((int) (*s & 0x0f) << 12) +
 	    ((*(s + 1) & 0x3f) << 6) + (*(s + 2) & 0x3f);
 	  s += 3;
 	}
+      else                     /* 4 byte */
+      {
+	  *us++ =
+	    ((int) (*s & 0x07) << 18) +
+	    ((*(s + 1) & 0x3f) << 12) +
+	    ((*(s + 2) & 0x3f) << 6) +
+	    ((*(s + 3) & 0x3f) << 0);
+	  s += 4;
+      }
       length++;
     }
   return (length);
--- a/bin/fribidi-main.c
+++ b/bin/fribidi-main.c
@@ -609,7 +609,7 @@ main (
 	    }
 
 	    if (*nl_found)
-	      printf (new_line);
+	      printf ("%s", new_line);
 	  }
       }
     }
--- a/charset/fribidi-char-sets.c
+++ b/charset/fribidi-char-sets.c
@@ -114,7 +114,7 @@ static FriBidiCharSetHandler char_sets[FRIBIDI_CHAR_SETS_NUM + 1] = {
 };
 
 #if FRIBIDI_USE_GLIB+0
-# include <glib/gstrfuncs.h>
+# include <glib.h>
 # define fribidi_strcasecmp g_ascii_strcasecmp
 #else /* !FRIBIDI_USE_GLIB */
 static char
--- a/lib/mem.h
+++ b/lib/mem.h
@@ -42,7 +42,7 @@
 #if FRIBIDI_USE_GLIB+0
 
 #ifndef __FRIBIDI_DOC
-# include <glib/gmem.h>
+# include <glib.h>
 #endif /* !__FRIBIDI_DOC */
 
 #define FriBidiMemChunk GMemChunk
--- a/bin/fribidi-main.c
+++ b/bin/fribidi-main.c
@@ -92,6 +92,7 @@ die2 (
 fribidi_boolean do_break, do_pad, do_mirror, do_reorder_nsm, do_clean;
 fribidi_boolean show_input, show_visual, show_basedir;
 fribidi_boolean show_ltov, show_vtol, show_levels;
+const int default_text_width = 80;
 int text_width;
 const char *char_set;
 const char *bol_text, *eol_text;
@@ -119,7 +120,8 @@ help (
   printf ("  -d, --debug           Output debug information\n"
 	  "  -t, --test            Test " FRIBIDI_NAME
 	  ", same as --clean --nobreak\n"
-	  "                        --showinput --reordernsm\n");
+	  "                        --showinput --reordernsm --width %d\n",
+	  default_text_width);
 #if FRIBIDI_MAIN_USE_ICONV_H+0
   printf ("  -c, --charset CS      Specify character set, default is %s.\n"
 	  "                        CS should be a valid iconv character set name\n",
@@ -136,7 +138,7 @@ help (
 	  "  -w, --width W         Screen width for padding, default is %d, but if\n"
 	  "                        environment variable COLUMNS is defined, its value\n"
 	  "                        will be used, --width overrides both of them.\n",
-	  text_width);
+	  default_text_width);
   printf
     ("  -B, --bol BOL         Output string BOL before the visual string\n"
      "  -E, --eol EOL         Output string EOL after the visual string\n"
@@ -215,7 +217,7 @@ main (
   char *s;
   FILE *IN;
 
-  text_width = 80;
+  text_width = default_text_width;
   do_break = true;
   do_pad = true;
   do_mirror = true;
@@ -324,6 +326,7 @@ main (
 	case 't':
 	  do_clean = show_input = do_reorder_nsm = true;
 	  do_break = false;
+	  text_width = default_text_width;
 	  break;
 	case 'c':
 	  char_set = my_fribidi_strdup (optarg);
