
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,7 +3,7 @@
 #AUTOMAKE_OPTIONS = 1.7 foreign dist-zip dist-bzip2
 AUTOMAKE_OPTIONS = foreign 1.11 dist-zip dist-xz
 
-SUBDIRS = m4 include vq lib test doc
+SUBDIRS = m4 include vq lib test
 
 if BUILD_EXAMPLES 
 SUBDIRS += examples 
--- a/Makefile.in
+++ b/Makefile.in
@@ -127,7 +127,7 @@
 	distdir dist dist-all distcheck
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = m4 include vq lib test doc examples
+DIST_SUBDIRS = m4 include vq lib test
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
