diff -NEbur '--exclude-from=/home/dbo/DDT/cdk/scripts/diff-exclude' nhd2-exp.org/lib/libeplayer3/container/container_ffmpeg.c nhd2-exp/lib/libeplayer3/container/container_ffmpeg.c
--- nhd2-exp.org/lib/libeplayer3/container/container_ffmpeg.c	2014-09-16 09:40:58.246413684 +0200
+++ nhd2-exp/lib/libeplayer3/container/container_ffmpeg.c	2014-10-13 12:54:35.239784005 +0200
@@ -935,7 +935,11 @@
 		* but the file is played back well. so remove this
 		* until other works are done and we can prove this.
 		*/
+#if LIBAVFORMAT_VERSION_MAJOR < 54
 		av_close_input_file(avContext);
+#else
+		avformat_close_input(&avContext);
+#endif
 		releaseMutex(FILENAME, __FUNCTION__,__LINE__);
 		return cERR_CONTAINER_FFMPEG_STREAM;
 #endif
@@ -1442,8 +1446,12 @@
 
 	if (avContext != NULL) 
 	{
+#if LIBAVFORMAT_VERSION_MAJOR < 54
 		av_close_input_file(avContext);
 		avContext = NULL;
+#else
+		avformat_close_input(&avContext);
+#endif
 	}
 
 	isContainerRunning = 0;
diff -NEbur '--exclude-from=/home/dbo/DDT/cdk/scripts/diff-exclude' nhd2-exp.org/lib/libeplayer3/tools/meta.c nhd2-exp/lib/libeplayer3/tools/meta.c
--- nhd2-exp.org/lib/libeplayer3/tools/meta.c	2014-09-16 09:40:58.190413682 +0200
+++ nhd2-exp/lib/libeplayer3/tools/meta.c	2014-10-13 12:54:35.239784005 +0200
@@ -63,12 +63,17 @@
 
         return -1;
     }
-
+#if LIBAVFORMAT_VERSION_MAJOR < 54
     if (av_find_stream_info(avContext) < 0) 
     {
         printf("Error av_find_stream_info\n");
     }
-
+#else
+    if (avformat_find_stream_info(avContext, NULL) < 0) 
+    {
+        printf("Error avformat_find_stream_info\n");
+    }
+#endif
     printf("\n***\n");
     dump_metadata();
      
