--- openssl-0.9.8g.orig/Configure	2013-02-05 12:58:59.000000000 +0100
+++ openssl-0.9.8g/Configure	2014-05-27 16:40:20.457317425 +0200
@@ -403,6 +403,12 @@
 "QNX6",       "gcc:-DTERMIOS::::-lsocket::${no_asm}:dlfcn:bsd-gcc-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 "QNX6-i386",  "gcc:-DL_ENDIAN -DTERMIOS -O2 -Wall::::-lsocket:${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:bsd-gcc-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 
+# Linux on SuperH
+"linux-sh","sh4-linux-gcc:-DL_ENDIAN -DTERMIO -pipe -Os::-D_REENTRANT::-ldl:BN_LLONG::::::::::::dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
+
+# Linux on ST200
+"linux-st200","st231-linux-gcc:-DL_ENDIAN -DTERMIO -O3::-D_REENTRANT::-ldl:BN_LLONG::::::::::::dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
+
 #### SCO/Caldera targets.
 #
 # Originally we had like unixware-*, unixware-*-pentium, unixware-*-p6, etc.
--- openssl-0.9.8l.old/Makefile.org	2009-03-03 23:40:29.000000000 +0100
+++ openssl-0.9.8l.new/Makefile.org	2014-02-13 17:14:20.564326000 +0100
@@ -132,7 +132,7 @@
 
 BASEADDR=
 
-DIRS=   crypto fips ssl engines apps test tools
+DIRS=   crypto ssl
 SHLIBDIRS= crypto ssl fips
 
 # dirs in crypto to build
--- openssl-0.9.8x.orig/Makefile.org    2010-01-27 17:06:36.000000000 +0100
+++ openssl-0.9.8x/Makefile.org 2014-08-24 13:35:14.335683341 +0200
@@ -615,7 +615,7 @@
 dist_pem_h:
        (cd crypto/pem; $(MAKE) -e $(BUILDENV) pem.h; $(MAKE) clean)

-install: all install_docs install_sw
+install: all install_sw

 install_sw:
        @$(PERL) $(TOP)/util/mkdir-p.pl $(INSTALL_PREFIX)$(INSTALLTOP)/bin \
